src/Symfony/Component/Process/Pipes/WindowsPipes.php

```diff
@@ -88,6 +91,16 @@ public function __construct($input, $haveReadSupport)
     public function __destruct()
     {
         $this->close();
+
+        foreach ($this->files as $file) {
```

Comment:
this should be removed: unlinking creates race conditions
there's no way to cleanup these

Reply:
I see, is there a simple scenario I can produce a race condition to test against? I'd like to attempt to cover a race condition if it exists, in addition to exploring your suggestion in the constructor comment.

I appreciate your input, thanks!

Reply:
mm dunno sorry, race conditions are the worst to test :)

Reply:
Absolutely! Allow me to revise my question: how does unlinking create a race condition?

WindowsPipes A will exclusively use and unlink sf_proc_00 files on destruct, while a concurrent WindowsPipes B instance will operate using sf_proc_01 files, C with sf_proc_02 files and so forth, a race condition between WindowsPipes seems improbable. 

I must be overlooking something, I'd like to know more about the race condition you are seeing to help develop a satisfactory fix for this issue and avoid creating similar race conditions in the future, thanks!

Reply:
e.g. unlink below can succeed between the fopen and flock of another process
or unlink can fail if the file is locked by a concurrent process.
All logic involving two functions (file_exists+unlink) can fail if anything happens between the calls.
We need to carefully review each of them to be sure this cannot lead to bad side effects (either failures or filling the directory with an ever growing number of files).

Did you consider my suggestion above?